File: advanced/graphql/news-graphql/src/main/java/org/tsdes/advanced/graphql/newsgraphql/MyGenericGraphQLError.java
package org.tsdes.advanced.graphql.newsgraphql;

import com.fasterxml.jackson.annotation.JsonIgnore;
import graphql.ErrorType;
import graphql.GraphQLError;
import graphql.language.SourceLocation;

import java.util.List;

/*
    TODO remove once Kotlin fixed

    Also check
    https://github.com/graphql-java/graphql-java/issues/1123
 */

public class MyGenericGraphQLError extends RuntimeException implements GraphQLError {


    public MyGenericGraphQLError(String message) {
        super(message);
    }

    @Override
    public String getMessage() {
        return super.getMessage();
    }

    @Override
    @JsonIgnore
    public List<SourceLocation> getLocations() {
        return null;
    }

    @Override
    @JsonIgnore
    public ErrorType getErrorType() {
        return null;
    }
}

File: advanced/graphql/news-graphql/src/main/kotlin/org/tsdes/advanced/graphql/newsgraphql/MyGenericGraphQLError.kt
package org.tsdes.advanced.graphql.newsgraphql

import com.fasterxml.jackson.annotation.JsonIgnore
import graphql.ErrorType
import graphql.GraphQLError
import graphql.language.SourceLocation
import java.lang.RuntimeException

/*
    FIXME

    TODO report issue to Kotlin

    Maybe related to following?
    https://youtrack.jetbrains.com/issue/KT-14115#tab=Comments
 */


//class MyGenericGraphQLError(message: String) : RuntimeException(message), GraphQLError {
//
//    override fun getMessage(): String? {
//        return super.message
//    }
//
//    @JsonIgnore
//    override fun getLocations(): List<SourceLocation>? {
//        return null
//    }
//
//    @JsonIgnore
//    override fun getErrorType(): ErrorType? {
//        return null
//    }
//}
